src/logic/add.rs

```diff
@@ -0,0 +1,265 @@
+use crate::data::{ApInt, DataAccessMut, ZipDataAccessMutSelf::{Inl, Ext}, Digit};
```

Comment:
Please consider renaming this file/module to `add_sub`.
I already see myself asking where we left our sub implementations.

Reply:
I have renamed it